src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/StatusUpdateService.kt

```diff
@@ -48,6 +52,16 @@ class StatusUpdateService(
         ValidatableActionResult.GeneralValidationError("The status ${statusUpdate.newStatus} is not valid"),
       )
 
+    var statusDetails: List<Cas2PersistedApplicationStatusDetail>? = null
```

Comment:
this adds to many 'returns' to the `create` method according to detekt, but not sure how to refactor it - from what I understand Kotlin doesn't like functions that could return more than one type, so how could I extract some of these returns to a separate method? 🤔

Reply:
added a suppression flag, following convo here https://mojdt.slack.com/archives/C03K0HB0LBE/p1707221283595289